internal/server/filesystem/server.go

```diff
@@ -119,115 +114,114 @@ func (s *Server) validatePathWindows(pathCtx internal.PathContext, path string)
-		}, err
+		return nil, err
 	}
 	err = s.hostAPI.Rmdir(request.Path, request.Force)
 	if err != nil {
 		klog.Errorf("failed Rmdir %v", err)
-		return &internal.RmdirResponse{
-			Error: err.Error(),
-		}, err
+		return nil, err
 	}
-	return &internal.RmdirResponse{
-		Error: "",
-	}, err
+	return nil, err
 }
-
 func (s *Server) LinkPath(ctx context.Context, request *internal.LinkPathRequest, version apiversion.Version) (*internal.LinkPathResponse, error) {
-	klog.V(4).Infof("calling LinkPath with targetPath %q sourcePath %q", request.TargetPath, request.SourcePath)
+	klog.V(2).Infof("Request: LinkPath with targetPath=%q sourcePath=%q", request.TargetPath, request.SourcePath)
+	createSymlinkRequest := &internal.CreateSymlinkRequest{
+		SourcePath: request.SourcePath,
+		TargetPath: request.TargetPath,
+	}
+	if _, err := s.CreateSymlink(ctx, createSymlinkRequest, version); err != nil {
+		klog.Errorf("Failed to forward to CreateSymlink: %v", err)
+		return nil, err
+	}
+	return &internal.LinkPathResponse{}, nil
+}
+
+func (s *Server) CreateSymlink(ctx context.Context, request *internal.CreateSymlinkRequest, version apiversion.Version) (*internal.CreateSymlinkResponse, error) {
+	klog.V(2).Infof("Request: CreateSymlink with targetPath=%q sourcePath=%q", request.TargetPath, request.SourcePath)
 	err := s.validatePathWindows(internal.POD, request.TargetPath)
 	if err != nil {
 		klog.Errorf("failed validatePathWindows for target path %v", err)
-		return &internal.LinkPathResponse{
-			Error: err.Error(),
-		}, err
+		return nil, err
 	}
 	err = s.validatePathWindows(internal.PLUGIN, request.SourcePath)
 	if err != nil {
 		klog.Errorf("failed validatePathWindows for source path %v", err)
-		return &internal.LinkPathResponse{
-			Error: err.Error(),
-		}, err
+		return nil, err
 	}
-	err = s.hostAPI.LinkPath(request.SourcePath, request.TargetPath)
-	errString := ""
+	err = s.hostAPI.CreateSymlink(request.SourcePath, request.TargetPath)
 	if err != nil {
-		klog.Errorf("failed LinkPath %v", err)
-		errString = err.Error()
+		klog.Errorf("failed CreateSymlink: %v", err)
+		return nil, err
 	}
-	return &internal.LinkPathResponse{
-		Error: errString,
-	}, err
+	return &internal.CreateSymlinkResponse{}, nil
 }
 
 func (s *Server) IsMountPoint(ctx context.Context, request *internal.IsMountPointRequest, version apiversion.Version) (*internal.IsMountPointResponse, error) {
-	klog.V(4).Infof("calling IsMountPoint with path %q", request.Path)
-	isMount, err := s.hostAPI.IsMountPoint(request.Path)
+	klog.V(2).Infof("Request: IsMountPoint with path=%q", request.Path)
+	isSymlinkRequest := &internal.IsSymlinkRequest{
+		Path: request.Path,
+	}
+	isSymlinkResponse, err := s.IsSymlink(ctx, isSymlinkRequest, version)
 	if err != nil {
-		klog.Errorf("failed IsMountPoint %v", err)
-		return &internal.IsMountPointResponse{
-			IsMountPoint: false,
-			Error:        err.Error(),
-		}, err
+		klog.Errorf("Failed to forward to IsSymlink: %v", err)
```

Comment:
"failed to forward ...

Reply:
What's the feedback here?

Reply:
oh, just small case in failed

Reply:
I think I set it as uppercase in many places, typically when we add logs statements they start with an uppercase letter right?

Reply:
I think the current format is lowercase. but definitely it is not consistent. 
if there are two many of them, we can use different PR for it.